lib/collection/src/collection/snapshots.rs

```diff
@@ -284,14 +290,24 @@ impl Collection {
             .await
     }
 
+    /// # Safety
+    ///
+    /// This function is *not* cancel-safe!
     pub async fn restore_shard_snapshot(
         &self,
         shard_id: ShardId,
         snapshot_path: &Path,
         this_peer_id: PeerId,
         is_distributed: bool,
         temp_dir: &Path,
+        cancel: cancel::CancellationToken,
     ) -> CollectionResult<()> {
+        // TODO:
```

Comment:
Is this TODO tracked elsewhere?

Reply:
I've created an issue for it here so that we don't lose track of it: https://github.com/qdrant/qdrant/issues/2913